server/issue.go

```diff
@@ -324,6 +330,24 @@ func (s *Server) composeIssueRelationship(ctx context.Context, issue *api.Issue)
 	return nil
 }
 
+// Only allow Owner/DBA as the assignee, not Developer.
+func (s *Server) validateAssigneeRoleByID(ctx context.Context, assigneeID int) error {
+	assignee, err := s.PrincipalService.FindPrincipal(ctx, &api.PrincipalFind{
+		ID: &assigneeID,
+	})
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find principal by id %d", assigneeID)).SetInternal(err)
+	}
+	if assignee == nil {
+		return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Principal ID not found: %d", assigneeID))
+	}
+	if assignee.Role != api.Owner && assignee.Role != api.DBA {
```

Comment:
System bot can also be allowed, this happens in the VCS triggered workflow.

Looks like we miss a test case here @d-bytebase ?

Reply:
We have a test case of TestVCS. I think this line plays the trick so that the test wasn't failing before.

https://github.com/bytebase/bytebase/blob/main/server/principal.go#L166

Reply:
oh, @boojack as @d-bytebase said https://github.com/bytebase/bytebase/blob/main/server/principal.go#L166 already does the trick to assign OWNER to the bot account, so we don't need to check BOT account separately.

Reply:
Removed.